plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -37,12 +38,118 @@ type X509Cert struct {
 	Timeout          config.Duration `toml:"timeout"`
 	ServerName       string          `toml:"server_name"`
 	ExcludeRootCerts bool            `toml:"exclude_root_certs"`
-	tlsCfg           *tls.Config
-	_tls.ClientConfig
+	Log              telegraf.Logger `toml:"-"`
+	commontls.ClientConfig
 	proxy.TCPProxy
+
+	tlsCfg    *tls.Config
 	locations []*url.URL
 	globpaths []*globpath.GlobPath
-	Log       telegraf.Logger
+}
+
+func (*X509Cert) SampleConfig() string {
+	return sampleConfig
+}
+
+func (c *X509Cert) Init() error {
+	// Check if we do have at least one source
+	if len(c.Sources) == 0 {
+		return errors.New("no source configured")
+	}
+
+	// Check the server name and transfer it if necessary
+	if c.ClientConfig.ServerName != "" && c.ServerName != "" {
+		return fmt.Errorf("both server_name (%q) and tls_server_name (%q) are set, but they are mutually exclusive", c.ServerName, c.ClientConfig.ServerName)
```

Comment:
This previously was not an error right?

Reply:
It was an error before, but checked in a different (now dissolved) [function](https://github.com/influxdata/telegraf/blob/6cfae2a1fd898cf7dfbca98565bcb534bf671050/plugins/inputs/x509_cert/x509_cert.go#L73).

Reply:
ah thank you